packages/x-data-grid-pro/src/hooks/features/dataSource/useGridDataSource.ts

```diff
@@ -52,28 +70,54 @@ export const useGridDataSource = (
     | 'sortingMode'
     | 'filterMode'
     | 'paginationMode'
+    | 'pageSizeOptions'
     | 'treeData'
+    | 'lazyLoading'
   >,
 ) => {
+  const setStrategyAvailability = React.useCallback(() => {
+    apiRef.current.setStrategyAvailability(
+      GridStrategyGroup.DataSource,
+      DataSourceRowsUpdateStrategy.Default,
+      props.unstable_dataSource && !props.lazyLoading ? () => true : () => false,
+    );
+  }, [apiRef, props.lazyLoading, props.unstable_dataSource]);
+
+  const [defaultRowsUpdateStrategyActive, setDefaultRowsUpdateStrategyActive] =
+    React.useState(false);
   const nestedDataManager = useLazyRef<NestedDataManager, void>(
     () => new NestedDataManager(apiRef),
   ).current;
+  const paginationModel = useGridSelector(apiRef, gridPaginationModelSelector);
   const groupsToAutoFetch = useGridSelector(apiRef, gridRowGroupsToFetchSelector);
   const scheduledGroups = React.useRef<number>(0);
+  const lastRequestIdentifier = React.useRef<string | null>(null);
+
   const onError = props.unstable_onDataSourceError;
 
+  const cacheChunkSize = React.useMemo(() => {
+    const sortedPageSizeOptions = props.pageSizeOptions
+      .map((option) => (typeof option === 'number' ? option : option.value))
+      .sort((a, b) => a - b);
+
+    return Math.min(paginationModel.pageSize, sortedPageSizeOptions[0]);
+  }, [paginationModel.pageSize, props.pageSizeOptions]);
```

Comment:
`pageSizeOptions` are usually passed directly in the prop which may lead to frequent recomputation of `cacheChunkSize`
`useLazyRef` will not reflect this change anyways IIMNM. How about moving this computation to the `useLazyRef` initializer?

Having said that, what if `pageSizeOptions` get updated and the user expects the `chunkSize to be updated` too? 🤔 
Should we reflect the updated value?

Reply:
I will move it to the `useLazyRef`. I think that we can start with having it not react to the changes. I don't see changing `pageSizeOptions` as a common use case.